hi-ml-multimodal/src/health_multimodal/image/model/model.py

```diff
@@ -147,84 +152,17 @@ def get_patchwise_projected_embeddings(self, input_img: torch.Tensor, normalize:
         return projected_embeddings
 
 
-class ImageEncoder(nn.Module):
-    """Image encoder trunk module for the ``ImageModel`` class.
-
-    :param img_model_type: Type of image model to use: either ``"resnet18"`` or ``"resnet50"``.
-    """
-
-    def __init__(self, img_model_type: str):
-        super().__init__()
-        self.img_model_type = img_model_type
-        self.encoder = self._create_encoder()
-
-    def _create_encoder(self, **kwargs: Any) -> nn.Module:
-        supported = ResnetType.RESNET18, ResnetType.RESNET50
-        if self.img_model_type not in supported:
-            raise NotImplementedError(f"Image model type \"{self.img_model_type}\" must be in {supported}")
-        encoder_class = resnet18 if self.img_model_type == ResnetType.RESNET18 else resnet50
-        encoder = encoder_class(pretrained=True, **kwargs)
-        return encoder
+class MultiImageModel(ImageModel):
+    def post_init_checks(self) -> None:
+        super().post_init_checks()
+        assert isinstance(self.encoder, MultiImageEncoder), "MultiImageModel only supports MultiImageEncoder"
```

Comment:
Can't we remove this method and simply adding an `__init__` and calling this line after calling `super()`'s `__init__`?

Reply:
updated.